modules/porous_flow/src/userobjects/PorousFlowDictator.C

```diff
@@ -126,6 +137,16 @@ PorousFlowDictator::porousFlowVariableNum(unsigned int moose_var_num) const
   return _pf_var_num[moose_var_num];
 }
 
+unsigned int
+PorousFlowDictator::mooseVariableNum(unsigned int porous_flow_var_num) const
+{
+  if (porous_flow_var_num >= _num_variables)
+    mooseError("The Dictator proclaims that there is no such PorousFlow variable with number ",
+               porous_flow_var_num,
+               ".  Exiting with error code 1984.");
```

Comment:
Nice! I hope you have a test for this ;-)

Reply:
Good point.   i've created some unit tests